includes/cli/class-import.php

```diff
@@ -32,26 +43,85 @@ class Import extends WP_CLI_Command {
 	 * @param array $assoc_args Assoc args.
 	 */
 	public function __invoke( $args, $assoc_args ) {
+		$ignore_terms = WP_CLI\Utils\get_flag_value( $assoc_args, 'ignore-terms', false );
+		$create_terms = WP_CLI\Utils\get_flag_value( $assoc_args, 'create-terms', false );
+		$file         = $args[0];
 
-		$file = $args[0];
 		if ( ! is_readable( $file ) ) {
-			WP_CLI::error( 'File not found or not readable.' );
+			WP_CLI::error( __( 'File not found or not readable.', 'newspack-popups' ) );
 		}
 
 		$data = wp_json_file_decode( $file, [ 'associative' => true ] ); // phpcs:ignore
 		if ( empty( $data ) ) {
-			WP_CLI::error( 'File is not valid JSON.' );
+			WP_CLI::error( __( 'File is not valid JSON.', 'newspack-popups' ) );
 		}
 
+		WP_CLI::success( __( 'JSON file loaded, initializing the importer...', 'newspack-popups' ) );
+
 		$importer = new Newspack_Popups_Importer( $data );
-		$result   = $importer->import();
+
+		$missing_terms = $importer->get_missing_terms_from_input();
+
+		if ( ! $ignore_terms && ! empty( $missing_terms ) ) {
+			foreach ( $missing_terms as $tax => $terms ) {
+				foreach ( $terms as $term ) {
+					$done = false;
+					// If term creation fails, or if the user cancels the selection, we will reset the strategy and ask again, that's why we have this while loop.
+					while ( ! $done ) {
+						$question = sprintf(
+							// translators: %1$s is the name of the term, %2$s is the taxonomy (category or tag).
+							__( 'The "%1$s" %2$s does not exist in this site. How do you want to proceed?', 'newspack-popups' ),
+							$term['name'],
+							$this->tax_label( $tax )
+						);
+
+						if ( ! $create_terms ) {
+							WP_CLI::warning( $question );
+							$strategy = $this->choose_term_strategy();
+						} else {
+							$strategy = 1;
+							$done     = true; // avoid infinte loop in case of error.
+						}
+
+						if ( 1 == $strategy ) { // Create term.
+
+							$new_id = $this->create_term( $term['name'], $tax );
+							if ( is_wp_error( $new_id ) ) {
+								WP_CLI::error( $new_id->get_error_message(), false );
+								// Dont set done and ask for the strategy again.
+							} else {
+								WP_CLI::success( __( 'Term created', 'newspack-popups' ) );
+								$done = true;
+							}
+						} elseif ( 2 == $strategy ) { // Map to existing term.
+
+							$new_id = $this->choose_existing_term( $tax );
+							if ( ! is_wp_error( $new_id ) ) {
+								WP_CLI::success( __( 'Term selected', 'newspack-popups' ) );
+								$importer->add_term_mapping( $term['id'], $new_id );
+								$done = true;
+							} // else means the user aborted the selection. Dont set done and ask for the strategy again.
+
+						} elseif ( 3 === $strategy ) { // Ignore term (do nothing).
+
+							WP_CLI::success( __( 'Ignoring', 'newspack-popups' ) );
+							$done = true;
+
+						}
+					}
+				}
+			}
+		}
+
+		WP_CLI::success( __( 'Running the importer', 'newspack-popups' ) );
```

Comment:
Shouldn't this just be a message?

```suggestion
		WP_CLI::log( __( 'Running the importer', 'newspack-popups' ) );
```